leshan-core/src/main/java/org/eclipse/leshan/core/node/codec/json/LwM2mNodeJsonEncoder.java

```diff
@@ -203,7 +203,7 @@ public void visit(LwM2mResourceInstance instance) {
 
             // create JSON resource element
             if (resource.isMultiInstances()) {
-                for (Entry<Integer, ?> entry : resource.getInstances().entrySet()) {
+                for (Entry<Integer, ?> entry : resource.getValues().entrySet()) {
```

Comment:
I would prefer we don't use deprecated `getValues()` (which will probably be deleted later)

Reply:
if we use getInstances, test json_encode_device_object_instance  failed, please suggest if getInstances() should be modified to be the same as getValues or code logic in JsonEncoder should be changed?

Reply:
logic in JsonEncoder should be adapted.

Reply:
Done

Reply:
I think we can avoid to create a new map :thinking:

Reply:
Agree, please review again

Reply:
This time you can avoid the cast because you know the map is a `Map<Integer, LwM2mResourceInstance>` (not `Map<Integer, ?>`

Reply:
Actually I thought about it , but then decided not to update map generics Type. But np, I'll change